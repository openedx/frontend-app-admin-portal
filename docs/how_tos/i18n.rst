####################
React App i18n HOWTO
####################

This document has moved to the frontend-platform repo: https://github.com/openedx/frontend-platform/blob/master/docs/how_tos/i18n.rst
