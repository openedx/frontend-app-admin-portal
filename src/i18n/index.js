import { messages as paragonMessages } from '@edx/paragon';

export default [
  paragonMessages,
];
