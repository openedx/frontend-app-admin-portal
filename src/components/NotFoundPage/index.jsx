import React from 'react';
import { Helmet } from 'react-helmet';
import { FormattedMessage } from '@edx/frontend-platform/i18n';

export const NotFound = () => (
  <>
    <Helmet>
      <title>Page Not Found</title>
    </Helmet>
    <div className="text-center py-5">
      <h1>404</h1>
      <p className="lead">
        <FormattedMessage
          id="admin.portal.not.found.page.message"
          defaultMessage="Oops, sorry we can't find that page!"
          description="The message displayed on the 404 page"
        />
      </p>
      <p>
        <FormattedMessage
          id="admin.portal.not.found.page.message2"
          defaultMessage="Either something went wrong or the page doesn't exist anymore."
          description="The message displayed on the 404 page"
        />
      </p>
    </div>
  </>
);

const NotFoundPage = () => (
  <main role="main">
    <div className="container-fluid mt-3">
      <NotFound />
    </div>
  </main>
);

export default NotFoundPage;
