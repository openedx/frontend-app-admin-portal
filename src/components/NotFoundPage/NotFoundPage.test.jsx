import React from 'react';
import renderer from 'react-test-renderer';
import { IntlProvider } from '@edx/frontend-platform/i18n';

import NotFoundPage from './index';

describe('<NotFoundPage />', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create((
        <IntlProvider locale="en">
          <NotFoundPage />
        </IntlProvider>
      ))
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
