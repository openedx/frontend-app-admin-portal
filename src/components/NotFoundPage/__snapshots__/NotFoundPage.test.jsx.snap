// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotFoundPage /> renders correctly 1`] = `
<main
  role="main"
>
  <div
    className="container-fluid mt-3"
  >
    <div
      className="text-center py-5"
    >
      <h1>
        404
      </h1>
      <p
        className="lead"
      >
        Oops, sorry we can't find that page!
      </p>
      <p>
        Either something went wrong or the page doesn't exist anymore.
      </p>
    </div>
  </div>
</main>
`;
